### PR TITLE
Mark deallocateAll as @nogc

### PR DESCRIPTION
**Pull request by edi33416**
> This is currently blocked by #5829 

Edit: This can be safely merged as #5829 is now merged.

### CODE REVIEW

**Comment by edi33416** [comment]
@andralex @ZombineDev @wilzbach

**Comment by andralex** [comment]
circleci crashes with no explanation, cc @wilzbach @MartinNowak

File: std/experimental/allocator/building_blocks/stats_collector.d [hunk 1]
```diff
@@ -742,3 +742,14 @@ public:
     // calls std.experimental.allocator.common.goodAllocSize
     assert((() pure nothrow @safe @nogc => a.goodAllocSize(1))());
 }
+
+// Test that deallocateAll infers from parent
+@system unittest
```
Comment (ZombineDev) [inline std/experimental/allocator/building_blocks/stats_collector.d]: Is there something that prevents marking the whole unittest additionally with `nothrow @nogc`?

File: std/experimental/allocator/building_blocks/scoped_allocator.d [hunk 2]
```diff
@@ -234,3 +238,15 @@ struct ScopedAllocator(ParentAllocator)
     assert(b.length == 42);
     () nothrow @nogc { a.deallocate(b); }();
 }
+
+// Test that deallocateAll infers from parent
+@system unittest
```
Comment (ZombineDev) [inline std/experimental/allocator/building_blocks/scoped_allocator.d]: `nothrow @nogc`?

File: std/experimental/allocator/building_blocks/region.d [hunk 3]
```diff
@@ -807,7 +821,7 @@ version(Posix) @system unittest
     version(OSX) {} else
     {
         assert((() nothrow @nogc => alloc.deallocate(b))());
-        assert(alloc.deallocateAll);
+        assert((() nothrow @nogc => alloc.deallocateAll())());
     }
```
Comment (ZombineDev) [inline std/experimental/allocator/building_blocks/region.d]: Put `nothrow @nogc` on the unittest?

File: std/experimental/allocator/building_blocks/region.d [hunk 4]
```diff
@@ -591,14 +602,16 @@ struct InSituRegion(size_t size, size_t minAlign = platformAlignment)
...
+    assert((() nothrow @nogc => r1.deallocate(a))());
+    assert((() nothrow @nogc => r1.deallocateAll())());
 
     InSituRegion!(65_536, 1024*4) r2;
     assert(r2.available <= 65_536);
     a = r2.allocate(2001);
     assert(a.length == 2001);
     const void[] buff = r2.allocate(42);
     assert((() nothrow @safe @nogc => r2.owns(buff))() == Ternary.yes);
+    assert((() nothrow @nogc => r2.deallocateAll())());
```
Comment (ZombineDev) [inline std/experimental/allocator/building_blocks/region.d]: ditto

File: std/experimental/allocator/building_blocks/region.d [hunk 5]
```diff
@@ -375,6 +376,16 @@ struct Region(ParentAllocator = NullAllocator,
...
 }
 
+@system unittest
+{
+    import std.experimental.allocator.mallocator : Mallocator;
+
+    auto reg = Region!(Mallocator)(1024 * 64);
+    const b = reg.allocate(101);
+    assert(b.length == 101);
+    assert((() nothrow @nogc => reg.deallocateAll())());
```
Comment (ZombineDev) [inline std/experimental/allocator/building_blocks/region.d]: ditto

File: std/experimental/allocator/building_blocks/quantizer.d [hunk 6]
```diff
@@ -277,3 +277,17 @@ struct Quantizer(ParentAllocator, alias roundingFunction)
...
+{
+    import std.experimental.allocator.building_blocks.region : Region;
+    alias MyAlloc = Quantizer!(Region!(),
+        (size_t n) => n.roundUpToMultipleOf(64));
+    testAllocator!(() => MyAlloc(Region!()(new ubyte[1024 * 64])));
+
+    auto a = MyAlloc(Region!()(new ubyte[1024 * 64]));
+    auto b = a.allocate(42);
+    assert(b.length == 42);
+    assert((() nothrow @nogc => a.deallocateAll())());
```
Comment (ZombineDev) [inline std/experimental/allocator/building_blocks/quantizer.d]: ditto

File: std/experimental/allocator/building_blocks/null_allocator.d [hunk 7]
```diff
@@ -77,8 +78,8 @@ struct NullAllocator
     assert(!NullAllocator.instance.expand(b, 42));
     assert(!NullAllocator.instance.reallocate(b, 42));
     assert(!NullAllocator.instance.alignedReallocate(b, 42, 0));
-    () nothrow @nogc { NullAllocator.instance.deallocate(b); }();
-    NullAllocator.instance.deallocateAll();
+    assert((() nothrow @nogc => NullAllocator.instance.deallocate(b))());
+    assert((() nothrow @nogc => NullAllocator.instance.deallocateAll())());
```
Comment (ZombineDev) [inline std/experimental/allocator/building_blocks/null_allocator.d]: ditto

File: std/experimental/allocator/building_blocks/free_list.d [hunk 8]
```diff
@@ -1223,7 +1236,7 @@ struct SharedFreeList(ParentAllocator,
     // Pull request #5556
     import std.experimental.allocator.mallocator : Mallocator;
     shared SharedFreeList!(Mallocator, chooseAtRuntime, 64) a;
-    scope(exit) a.deallocateAll;
+    scope(exit) assert((() nothrow @nogc => a.deallocateAll())());
```
Comment (ZombineDev) [inline std/experimental/allocator/building_blocks/free_list.d]: ditto

**Pull request merged**